simulation/traffic_simulator/include/traffic_simulator/simulation_clock/simulation_clock.hpp

```diff
@@ -23,36 +23,39 @@ namespace traffic_simulator
 class SimulationClock : rclcpp::Clock
 {
 public:
-  explicit SimulationClock(double realtime_factor, double frame_rate);
+  explicit SimulationClock(bool use_sim_time, double realtime_factor, double frame_rate);
 
   auto getCurrentRosTime() -> rclcpp::Time;
 
   auto getCurrentRosTimeAsMsg() -> rosgraph_msgs::msg::Clock;
 
-  auto getCurrentScenarioTime() const { return (frame_ - frame_offset_) / frame_rate; }
+  auto getCurrentScenarioTime() const { return time_ - time_offset_; }
 
-  auto getCurrentSimulationTime() const { return frame_ / frame_rate; }
+  auto getCurrentSimulationTime() const { return time_; }
 
-  auto getStepTime() const { return realtime_factor / frame_rate; }
+  auto getStepTime() const { return realtime_factor_ / frame_rate_; }
 
   auto start() -> void;
 
-  auto started() const { return not std::isnan(frame_offset_); }
+  auto started() const { return not std::isnan(time_offset_); }
 
   auto update() -> void;
 
-  const bool use_raw_clock;
+  auto setRealTimeFactor(double realtime_factor) { realtime_factor_ = realtime_factor; };
 
-  double realtime_factor;
+  const bool use_raw_clock_;
 
-  double frame_rate;
+  double realtime_factor_;
```

Comment:
Public data members are externally accessible, so don't add ugly underscores to variable name suffixes.

```suggestion
  double realtime_factor;
```